### PR TITLE
Map Fragment Layout

### PR DESCRIPTION
**Pull request by HoangChu-Claremont**
I have finished the layout and populated a sample when user clicking on a featured path. I will do the getting user's current location and populating appropriate images for each destination and featured paths tomorrow.

https://user-images.githubusercontent.com/65592963/176105677-0029a72b-1f53-480d-b06b-cc025120eb07.mp4

https://user-images.githubusercontent.com/65592963/176492582-13c7ac36-bb16-453e-8b53-76b26eed8d31.mp4

### CODE REVIEW

File: app/src/main/java/com/example/where_to_go/fragments/MapFragment.java [hunk 1]
```diff
@@ -52,23 +56,39 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+                googleMap.addMarker(markerOptions);
+            });
+        });
+        Log.i(TAG, "Map Created");
 
         // Featured Destination
-//        getFilteredDestination();
-//        setFilteredDestinationRecyclerView();
+        getFilteredDestination();
+        setFilteredDestinationRecyclerView();
```
Comment (jinsoo601) [inline app/src/main/java/com/example/where_to_go/fragments/MapFragment.java:91]: hmmmmmm. On line 117, you are calling `filteredDestinationAdapter.notifyDataSetChanged()` in UI thread. `filteredDestinationAdapter` is initialized in `setFilteredDestinationRecyclerView`. Normally this wouldn't be a problem bc `setFilteredDestinationRecyclerView` will run before the callback resolves, but there is no guarantee. I suggest switching the order of `setFilteredDestinationRecyclerView` and `getFilteredDestination`. In other words, set up your recycler view (data receiver) first, and then call your data getter, so we know for sure receiver is ready to get the data. You may have to move initialization of `filteredDestinations` to `setFilteredDestinationRecyclerView` to do this.

Comment (HoangChu-Claremont) [reply app/src/main/java/com/example/where_to_go/fragments/MapFragment.java:91]: If I switched getFilteredDestination() going after setFilteredDestinationRecyclerView(), the terminal says getItemCount() in setFilteredDestinationRecyclerView() returns null, which I think it's due to that we need a full data before putting it into the adapter before inflating RecylerView and set the adapter.

Comment (kvn93) [reply app/src/main/java/com/example/where_to_go/fragments/MapFragment.java:91]: That might be cause by the `filteredDestinations` list being initialize after rv tries to show the content from the list which would be null at that time. Try setting it as empty list as default value.

File: app/src/main/java/com/example/where_to_go/fragments/MapFragment.java [hunk 2]
```diff
@@ -52,34 +60,37 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
-//        getFilteredDestination();
-//        setFilteredDestinationRecyclerView();
+        setFilteredDestinationRecyclerView();
+
+        // Get a handle to the fragment and register the callback.
+        SupportMapFragment supportMapFragment = (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.google_map);
+        // When Google Map is loaded, test that we captured the fragment
+        assert supportMapFragment != null;
+
+        supportMapFragment.getMapAsync( googleMap -> {
```
Comment (jinsoo601) [inline app/src/main/java/com/example/where_to_go/fragments/MapFragment.java:79]: remove space before googleMap

File: app/src/main/java/com/example/where_to_go/fragments/MapFragment.java [hunk 3]
```diff
@@ -88,13 +99,17 @@ public void onResponse(@NonNull Call call, @NonNull Response response) {
...
+                    List<Destination> filteredResults = FilterAlgorithm.getTopRatedPath(jsonResults);
+                    filteredDestinations.addAll(filteredResults);
 
                     // Avoid the "Only the original thread that created a view hierarchy
                     // can touch its views adapter" error
                     ((Activity) requireContext()).runOnUiThread(() -> {
-                        //change View Data
+                        // Update the Adapter
                         filteredDestinationAdapter.notifyDataSetChanged();
+                        // Set the Map
```
Comment (jinsoo601) [inline app/src/main/java/com/example/where_to_go/fragments/MapFragment.java:110]: unnecessary comment

**Review by jinsoo601 — APPROVED** [review]
Two minor comments, looks good!

**Review by jinsoo601 — APPROVED** [review]
Looks good to me! Can you just add a video of drag&drop feature in PR Description before merging?

**Comment by HoangChu-Claremont** [comment]
Updated: add a gif for drag-and-drop test

**Pull request merged**